libs/parallelism/algorithms/include/hpx/parallel/algorithms/search.hpp

```diff
@@ -364,110 +453,138 @@ namespace hpx { namespace parallel { inline namespace v1 {
-                                    traits::projected<Proj2, FwdIter2>>::value)>
-    typename util::detail::algorithm_result<ExPolicy, FwdIter>::type search_n(
-        ExPolicy&& policy, FwdIter first, std::size_t count, FwdIter2 s_first,
-        FwdIter2 s_last, Pred&& op = Pred(), Proj1&& proj1 = Proj1(),
-        Proj2&& proj2 = Proj2())
+        typename Pred = parallel::v1::detail::equal_to,
+        typename Proj1 = parallel::util::projection_identity,
+        typename Proj2 = parallel::util::projection_identity,
+        HPX_CONCEPT_REQUIRES_(hpx::is_execution_policy<
+            ExPolicy>::value&& hpx::traits::is_iterator<FwdIter>::value&&
+                parallel::traits::is_projected<Proj1, FwdIter>::value&&
+                    hpx::traits::is_iterator<FwdIter2>::value&&
+                        parallel::traits::is_projected<Proj2, FwdIter2>::value&&
+                            parallel::traits::is_indirect_callable<ExPolicy,
+                                Pred,
+                                parallel::traits::projected<Proj1, FwdIter>,
+                                parallel::traits::projected<Proj2,
+                                    FwdIter2>>::value)>
+    HPX_DEPRECATED_V(1, 6, "Please use hpx::search_n instead.")
+    typename parallel::util::detail::algorithm_result<ExPolicy, FwdIter>::type
+        search_n(ExPolicy&& policy, FwdIter first, std::size_t count,
+            FwdIter2 s_first, FwdIter2 s_last, Pred&& op = Pred(),
+            Proj1&& proj1 = Proj1(), Proj2&& proj2 = Proj2())
     {
         static_assert((hpx::traits::is_forward_iterator<FwdIter>::value),
             "Requires at least forward iterator.");
         static_assert((hpx::traits::is_forward_iterator<FwdIter2>::value),
             "Subsequence requires at least forward iterator.");
 
-        typedef hpx::is_sequenced_execution_policy<ExPolicy> is_seq;
+        using is_seq = hpx::is_sequenced_execution_policy<ExPolicy>;
 
-        return detail::search_n<FwdIter>().call(std::forward<ExPolicy>(policy),
-            is_seq(), first, count, s_first, s_last, std::forward<Pred>(op),
-            std::forward<Proj1>(proj1), std::forward<Proj2>(proj2));
+        return detail::search_n<FwdIter, FwdIter>().call(
+            std::forward<ExPolicy>(policy), is_seq(), first, count, s_first,
+            s_last, std::forward<Pred>(op), std::forward<Proj1>(proj1),
+            std::forward<Proj2>(proj2));
     }
 }}}    // namespace hpx::parallel::v1
+
+namespace hpx {
+
+    HPX_INLINE_CONSTEXPR_VARIABLE struct search_t final
+      : hpx::functional::tag<search_t>
+    {
+    private:
+        // clang-format off
+        template <typename FwdIter, typename FwdIter2,
+            typename Pred = parallel::v1::detail::equal_to,
+            HPX_CONCEPT_REQUIRES_(
+                traits::is_forward_iterator<FwdIter>::value &&
+                traits::is_forward_iterator<FwdIter2>::value
```

Comment:
Can we add `&& is_invocable_v<Pred, typename std::iterator_traits<FwdIter1>::value_type, typename std::iterator_traits<FwdIter1>::value_type>` (or do we have a more appropriate trait for that)? Same in the other `HPX_CONCEPT_REQUIRES_`.

Reply:
Yes totally, forgot to add them back sorry